### PR TITLE
test(controller): run shell scripts through shellcheck

### PR DESCRIPTION
**Pull request by Joshua-Anderson**

### CODE REVIEW

File: controller/bin/boot [hunk 1]
```diff
@@ -16,18 +16,20 @@ export ETCD_PATH=${ETCD_PATH:-/deis/controller}
...
-until etcdctl --no-sync -C $ETCD ls >/dev/null 2>&1; do
+until etcdctl --no-sync -C "$ETCD" ls >/dev/null 2>&1; do
 	echo "waiting for etcd at $ETCD..."
-	sleep $(($ETCD_TTL/2))  # sleep for half the TTL
+	sleep $((ETCD_TTL/2))  # sleep for half the TTL
 done
 
 # wait until etcd has discarded potentially stale values
-sleep $(($ETCD_TTL+1))
+sleep $((ETCD_TTL+1))
```
Comment (carmstrong) [inline controller/bin/boot]: This looks funky to me - my brain wants to ask "are you sure ETCD_TTL is really a variable and not implicitly zero??", but I have to trust shellcheck.

File: controller/bin/boot [hunk 2]
```diff
@@ -16,18 +16,20 @@ export ETCD_PATH=${ETCD_PATH:-/deis/controller}
...
 
 function etcd_set_default {
 	set +e
-	ERROR="$(etcdctl --no-sync -C $ETCD mk $ETCD_PATH/$1 $2 2>&1 >/dev/null)"
-	if [[ $? -ne 0 && $(echo $ERROR | grep -ive "key already exists") ]]; then
+	ERROR="$(etcdctl --no-sync -C "$ETCD" mk "$ETCD_PATH/$1" "$2" 2>&1 >/dev/null)"
+	# We're not compairing output with -n so error doesn't apply
```
Comment (carmstrong) [inline controller/bin/boot]: spelling typo

File: controller/bin/boot [hunk 3]
```diff
@@ -37,8 +39,10 @@ function etcd_set_default {
 
 function etcd_safe_mkdir {
 	set +e
-	ERROR="$(etcdctl --no-sync -C $ETCD mkdir $1 2>&1 >/dev/null)"
-	if [[ $? -ne 0 && $(echo $ERROR | grep -ive "key already exists") ]]; then
+	ERROR="$(etcdctl --no-sync -C "$ETCD" mkdir "$1" 2>&1 >/dev/null)"
+	# We're not compairing output with -n so error doesn't apply
```
Comment (carmstrong) [inline controller/bin/boot]: Spelling typo

File: controller/bin/boot (hunk 1 above)
Comment (Joshua-Anderson) [inline controller/bin/boot]: Shellcheck doesn't want arithmetic variables to use `$`

**Comment by mboersma** [comment]
Code LGTM but for the two typos already identified.

**Comment by Joshua-Anderson** [comment]
Fixed spelling errors.

**Comment by Joshua-Anderson** [comment]
I fixed the underling cause of the shellcheck warning, so the comment is no longer necessary.

**Comment by bacongobbler** [comment]
code LGTM

**Pull request merged**